File: server/src/main/java/com/fixsync/server/service/RepairItemService.java
package com.fixsync.server.service;

import com.fixsync.server.dto.request.RepairItemRequest;
import com.fixsync.server.dto.response.PageResponse;
import com.fixsync.server.dto.response.RepairItemResponse;
import org.springframework.data.domain.Pageable;

import java.util.List;
import java.util.UUID;

public interface RepairItemService {
    RepairItemResponse createRepairItem(RepairItemRequest request);
    RepairItemResponse updateRepairItem(UUID id, RepairItemRequest request);
    RepairItemResponse getRepairItemById(UUID id);
    List<RepairItemResponse> getRepairItemsByDeviceId(UUID deviceId);
    PageResponse<RepairItemResponse> getRepairItemsByDeviceId(UUID deviceId, Pageable pageable);
    PageResponse<RepairItemResponse> getAllRepairItems(Pageable pageable);
    void deleteRepairItem(UUID id);
    Integer calculateTotalCostByDeviceId(UUID deviceId);
}

File: client/src/services/repairItem.service.ts
import apiClient from "./api";
import { API_ENDPOINTS } from "../constants";
import type { ApiResponse, PageResponse, RepairItem } from "../types";

export const repairItemService = {
  /**
   * Get repair items by device with pagination (server supports paginated flag)
   */
  getRepairItemsByDevice: async (
    deviceId: string,
    page: number = 0,
    size: number = 10,
    sortBy: string = "createdAt",
    sortDir: string = "DESC"
  ): Promise<PageResponse<RepairItem>> => {
    const response = await apiClient.get<ApiResponse<PageResponse<RepairItem>>>(
      API_ENDPOINTS.REPAIR_ITEMS.BY_DEVICE(deviceId),
      {
        params: { paginated: true, page, size, sortBy, sortDir },
      }
    );
    return response.data.data;
  },

  /**
   * Get all repair items with pagination
   */
  getAllRepairItems: async (
    page: number = 0,
    size: number = 10,
    sortBy: string = "createdAt",
    sortDir: string = "DESC"
  ): Promise<PageResponse<RepairItem>> => {
    const response = await apiClient.get<ApiResponse<PageResponse<RepairItem>>>(
      API_ENDPOINTS.REPAIR_ITEMS.BASE,
      {
        params: { page, size, sortBy, sortDir },
      }
    );
    return response.data.data;
  },

  /**
   * Get repair item detail
   */
  getRepairItemById: async (id: string): Promise<RepairItem> => {
    const response = await apiClient.get<ApiResponse<RepairItem>>(
      API_ENDPOINTS.REPAIR_ITEMS.BY_ID(id)
    );
    return response.data.data;
  },

  /**
   * Create repair item
   */
  createRepairItem: async (
    data: Omit<RepairItem, "id" | "createdAt">
  ): Promise<RepairItem> => {
    const response = await apiClient.post<ApiResponse<RepairItem>>(
      API_ENDPOINTS.REPAIR_ITEMS.BASE,
      data
    );
    return response.data.data;
  },

  /**
   * Update repair item
   */
  updateRepairItem: async (
    id: string,
    data: Omit<RepairItem, "id" | "createdAt">
  ): Promise<RepairItem> => {
    const response = await apiClient.put<ApiResponse<RepairItem>>(
      API_ENDPOINTS.REPAIR_ITEMS.BY_ID(id),
      data
    );
    return response.data.data;
  },

  /**
   * Delete repair item
   */
  deleteRepairItem: async (id: string): Promise<void> => {
    await apiClient.delete(API_ENDPOINTS.REPAIR_ITEMS.BY_ID(id));
  },
};


